src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -927,6 +955,23 @@ DeferErrorIfQueryNotSupported(Query *queryTree)
 		errorMessage = "subquery in OFFSET is not supported in multi-shard queries";
 	}
 
+	List *rteList = ExtractRangeTableEntryList(queryTree);
+	if (CheckRTEList(rteList, IsCitusLocalTableRTE))
```

Comment:
Again, why don't we have Postgres local tables in this check?  Also, why not using `FindNodeMatchingCheckFunction`?

Reply:
let me give it a try

Reply:
it works fine, but the only thing seems suspicious to me is https://github.com/citusdata/citus/pull/4133/commits/d52f4a9a90d12f6ba88e6fbd289fd0954b52687d#diff-c3c8cc3c1e44703ca8536714510997beR420-R421